menpo/model/pca.py

```diff
@@ -1566,3 +1674,114 @@ def __str__(self):
+                   default: ''
+        size: tuple of two ints, the size of the image,
+              default:(600, 600)
+        list_weights: list of weights that std will be multipied with
+                      default: [-2, 2]
+        list_components : int, list or ndarray of the components to be rendered
+                         default: np.arange(5)
+        bgcolor : the background color of the rendering
+        camera_settings : camera settings of the rendering
+        mesh_color : tuple of three floats between 0-1 that defines mesh color
+        """
+        try:
+            from mayavi import mlab
+        except ImportError:
+            print('Cannot import mlab')
+            return 1
+
+        if export_path is None:
+            export_path = Path.cwd()
+        elif isinstance(export_path, str):
+            export_path = Path(export_path)
+
+        if not export_path.exists():
+            try:
+                export_path.mkdir(parents=True)
+            except OSError:
+                print('Cannot create directory')
+                return -1
+
+        fig = mlab.figure(bgcolor=bgcolor, size=size)
+        if camera_settings is not None:
+            mlab.move(*camera_settings[0])
+            mlab.view(*camera_settings[1])
+            mlab.roll(camera_settings[2])
+        else:
+            fig.scene.z_plus_view()
+
+        if isinstance(list_weights, int):
+            list_weights = [list_weights]
+
+        if isinstance(list_components, int):
+            list_components = [list_components]
+        list_components = np.asarray(list_components)
+        are_components = list_components < self.n_components
+        components_to_be_rendered = list_components[are_components]
+        for not_exist_component in list_components[~are_components]:
+            print('Component {} does not exist'.format(not_exist_component))
+
+        mesh = self.mean()
+        s = mlab.triangular_mesh(mesh.points[:, 0], mesh.points[:, 1],
+                                 mesh.points[:, 2], mesh.trilist,
+                                 color=mesh_color)
+
+        mlab.savefig(str(export_path / 'Mean face.{}'.format('png')))
+        for component in components_to_be_rendered:
+            parameters = np.zeros(component+1)
+            std = self.eigenvalues[component] ** 0.5
+            for weight in list_weights:
+                parameters[component] = std*weight
+                s.mlab_source.points = self.instance(parameters).points
+                full_filename = '{}_{}_{}.{}'.format(filename,
+                                                     component,
+                                                     weight,
+                                                     'png')
+                mlab.savefig(str(export_path / full_filename))
+
+    def to_list(self):
+        r"""
+        Returns the main attributes of a model as a list
+        for serialization with avatars.serialize_morphable_model_to_binary
```

Comment:
What function is `avatars.serialize_morphable_model_to_binary`? Presumably an internal function from your fork?

Reply:
Yup, I will remove it.